src/ol_infrastructure/applications/mitlearn/__main__.py

```diff
@@ -1584,10 +1246,611 @@
+        external_hostname=mitlearn_config.require("heroku_domain"),
+    ),
+    opts=ResourceOptions(
+        aliases=[
+            Alias(
+                f"ol-mitlearn-external-service-apisix-upstream-{stack_info.env_suffix}",
+            )
+        ],
+        delete_before_replace=True,
+    ),
+)
+
+# Actions specific to non-Kubernetes (Heroku) deployments
+if not mitlearn_config.get_bool("k8s_deploy"):
+    # Create the OIDC secret needed for APISIX when deployed outside K8s
+    oidc_secret_name, oidc_secret = create_oidc_k8s_secret(
+        stack_info=stack_info,
+        namespace=learn_namespace,
+        labels=application_labels,
+        vault_k8s_resources=vault_k8s_resources,
+        opts=ResourceOptions(
+            delete_before_replace=True,
+            parent=vault_k8s_resources,
+            depends_on=[mitlearn_vault_static_secrets],
+        ),
+    )
+
+    # We need to be able to change `unauth_action` depending on the route but otherwise
+    # the settings for the oidc plugin will be unchanged
+    # TODO (mmd, YYYY-MM-DD): Leaving this out of OLApisix* refactoring
+    legacy_base_oidc_plugin_config = {
+        "scope": "openid profile email",
+        "bearer_only": False,
+        "introspection_endpoint_auth_method": "client_secret_basic",
+        "ssl_verify": False,
+        "logout_path": "/logout/oidc",
+        "post_logout_redirect_uri": f"https://{mit_learn_api_domain}/logout/",
+    }
+
+    # MITx Online doesn't have a CI environment. Remove this once that ceases to be true.
+    # TODO(TMM 2025-04-04): Remove this hack once there is a CI deployment of Learn
+    mitxonline_consul_opts = get_consul_provider(
+        stack_info,
+        consul_address=f"https://consul-mitxonline-{stack_info.env_suffix}.odl.mit.edu",
+        provider_name=f"consul-provider-mitxonline-{stack_info.env_suffix}",
+    )
+    xpro_consul_opts = get_consul_provider(
+        StackInfo(
+            name="CI",
+            namespace=stack_info.namespace,
+            env_suffix="ci",
+            env_prefix=stack_info.env_prefix,
+            full_name=stack_info.full_name,
+        ),
+        consul_address=f"https://consul-xpro-{stack_info.env_suffix}.odl.mit.edu",
+        provider_name=f"consul-provider-xpro-{stack_info.env_suffix}",
+    )
+    consul.Keys(
+        "learn-api-domain-consul-key-for-mitxonline-openedx",
+        keys=[
+            consul.KeysKeyArgs(
+                path="edxapp/learn-api-domain",
+                delete=True,
+                value=mit_learn_api_domain,
+            )
+        ],
+        opts=mitxonline_consul_opts,
+    )
+    consul.Keys(
+        f"learn-api-domain-consul-key-for-xpro-openedx-{stack_info.env_suffix}",
+        keys=[
+            consul.KeysKeyArgs(
+                path="edxapp/learn-api-domain",
+                delete=True,
+                value=mit_learn_api_domain,
+            )
+        ],
+        opts=xpro_consul_opts,
+    )
+
```

Comment:
The xPRO consul keys block should move out of the `CI` conditional now that we've got a CI learn environment.